### PR TITLE
add new parent exception

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

add new parent exception which has  detailmessage.
suppressed the sonar warning  "Inheritance tree of classes should not be too deep".
better to talk with team and create another ticket to tidy up exceptions. 
to keep it simple this can be the first pr.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipFileProcessingFailedException.java [hunk 1]
```diff
@@ -3,7 +3,7 @@
 /**
  * Base class for all exceptions that represent failed zip file processing situation.
  */
-public abstract class ZipFileProcessingFailedException extends RuntimeException {
+public abstract class ZipFileProcessingFailedException extends RejectionException {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipFileProcessingFailedException.java:6]: This one also needs to be an interface

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package uk.gov.hmcts.reform.bulkscanprocessor.exceptions;
 
-public class ZipNameNotMatchingMetaDataException extends InvalidMetafileException {
+public class ZipNameNotMatchingMetaDataException extends RejectionException implements InvalidMetafileException {
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java:3]: All exceptions that previously extended `InvalidMetadatafileException` should now extend `InvalidEnvelopeException`. It, in turn, along with `PaymentsDisabledException` and `ServiceDisabledException`, should extend `RejectionException`.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java:3]: so InvalidEnvelopeException will be interface?

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java:3]: On a second thought, maybe we could get rid of `InvalidEnvelopeException` and just replace it with `RejectionException`. Technically, it doesn't add any benefit.

Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/exceptions/ZipNameNotMatchingMetaDataException.java:3]: > so InvalidEnvelopeException will be interface?

I think it should be completely replaced by `RejectionException`.

**Pull request closed without merging**